pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -87,11 +87,8 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 			return toAdmissionResponseError(err)
 		}
 		if cc.GetDataVolumeTTLSeconds(config) >= 0 {
-			if modifiedDataVolume.Annotations == nil {
-				modifiedDataVolume.Annotations = make(map[string]string)
-			}
 			if modifiedDataVolume.Annotations[cc.AnnDeleteAfterCompletion] != "false" {
-				modifiedDataVolume.Annotations[cc.AnnDeleteAfterCompletion] = "true"
+				cc.AddAnnotation(modifiedDataVolume, cc.AnnDeleteAfterCompletion, "true")
```

Comment:
Maybe worth a comment as well, similar to `const defaultDataVolumeTTLSeconds = -1`
as these are the GC defaulting points